File: src/test/java/net/pterodactylus/rhynodge/filters/comics/HeldentageFilterTest.java
package net.pterodactylus.rhynodge.filters.comics;

import java.io.IOException;

import net.pterodactylus.rhynodge.filters.ResourceLoader;
import net.pterodactylus.rhynodge.states.ComicState;
import net.pterodactylus.rhynodge.states.HtmlState;

import org.jsoup.nodes.Document;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.contains;

/**
 * Unit test for {@link HeldentageFilter}.
 *
 * @author <a href="mailto:bombe@pterodactylus.net">David ‘Bombe’ Roden</a>
 */
public class HeldentageFilterTest {

	private final HeldentageFilter heldentageFilter = new HeldentageFilter();
	private final HtmlState htmlState;

	public HeldentageFilterTest() throws IOException {
		Document document = ResourceLoader.loadDocument(HeldentageFilter.class, "heldentage.html", "http://www.der-flix.de/");
		htmlState = new HtmlState("http://www.der-flix.de/", document);
	}

	@Test
	public void comicIsParsedCorrectly() {
		ComicState comicState = (ComicState) heldentageFilter.filter(htmlState);
		assertThat(comicState.comics(), contains(
				ComicMatchers.isComic("", contains(
						ComicMatchers.isStrip("http://www.der-flix.de/images/heldentage/Tag_916.jpg", "")
				))
		));
	}

}

File: src/test/kotlin/net/pterodactylus/rhynodge/filters/comics/HeldentageFilterTest.kt
package net.pterodactylus.rhynodge.filters.comics

import net.pterodactylus.rhynodge.filters.ResourceLoader
import net.pterodactylus.rhynodge.states.HtmlState
import net.pterodactylus.rhynodge.states.ComicState
import net.pterodactylus.rhynodge.states.ComicState.Comic
import net.pterodactylus.rhynodge.states.ComicState.Strip
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.contains
import org.junit.Test

/**
 * Unit test for [HeldentageFilter].
 */
class HeldentageFilterTest {

	private val heldentageFilter = HeldentageFilter()
	private val htmlState = ResourceLoader
		.loadDocument(HeldentageFilter::class.java, "heldentage.html", "http://www.der-flix.de/")
		.let { HtmlState("http://www.der-flix.de/", it) }

	@Test
	fun comicIsParsedCorrectly() {
		val comicState = heldentageFilter.filter(htmlState) as ComicState
		assertThat(comicState.comics(), contains(
				Comic("Glückskind").add(Strip("http://www.der-flix.de/media/djmediatools/cache/1-glueckskind/1500x275-towidth-100-glueckskind_301.jpg", ""))
		))
	}

}
